### PR TITLE
Second take for #536 "configure compiled container parent class"

### LINKED ISSUE #535: Allow extending \DI\CompiledContainer
I'm currently stuck with a problem with the compiled container:

I've made an extension over \DI\Container to access array entries by dot notation. Thanks to the ability to [set container class](https://github.com/PHP-DI/PHP-DI/blob/master/src/ContainerBuilder.php#L103) on ContainerBuilder it is a breeze to use my modified container

The problem comes when I compile the container (definitions), even though [compiled container template](https://github.com/PHP-DI/PHP-DI/blob/master/src/Compiler/Template.php#L4) allows to set file/class name it still hardcodes \DI\ContainerInterface as the parent class to it

It would be great not to [override container class](https://github.com/PHP-DI/PHP-DI/blob/master/src/ContainerBuilder.php#L187) but to set both compiled class name and compiled class parent name

Unless I'm missing something this should be a simple/quick PR

Edit: typos

### PR DESCRIPTION
**Pull request by mnapoli**
Fix #535

ping @juliangut 

This PR includes #536 + tests.

### CODE REVIEW

File: src/Compiler.php [hunk 1]
```diff
@@ -56,8 +61,13 @@ class Compiler
      *
      * @return string The compiled container file name.
      */
-    public function compile(DefinitionSource $definitionSource, string $directory, string $className, bool $autowiringEnabled) : string
-    {
+    public function compile(
+        DefinitionSource $definitionSource,
+        string $directory,
+        string $className,
+        string $parentClassName,
```
Comment (bradynpoulsen) [inline src/Compiler.php]: This is a breaking change in the API. Is method intend to be internal or public?

File: src/Compiler/Template.php [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /**
  * This class has been auto-generated by PHP-DI.
  */
-class <?=$this->containerClass; ?> extends \DI\CompiledContainer
+class <?=$this->containerClass; ?> extends <?=$this->containerParentClass; ?>
```
Comment (bradynpoulsen) [inline src/Compiler/Template.php]: Are compiled containers supposed to honor the Container Interface? If so, it seems like allowing a custom parent should either validate such either in the compile/enableCompilation methods or here via `implements`

File: src/Compiler.php (hunk 1 above)
Comment (mnapoli) [reply src/Compiler.php]: No problem with BC here:

- the class was added in 6.0 and we are still in the alpha release cycle
- the class is not tagged with `@api` so it's an internal class

File: src/Compiler/Template.php (hunk 2 above)
Comment (mnapoli) [reply src/Compiler/Template.php]: Mmh this is a good question… Honestly I don't really mind what happens if someone decides to do something weird here?

Comment (juliangut) [reply src/Compiler/Template.php]: Interesting point here, I already considered this and I'm checking if the class [extends from \DI\CompiledContainer](https://github.com/juliangut/slim-php-di/blob/2.x/src/Configuration.php#L318)

Anyway I'm with @mnapoli, if you use any other thing than a ContainerInterface ... it's just a bug, fix it

**Pull request merged**